### PR TITLE
scripts/dts cleanups, part 4

### PR DESCRIPTION
**Pull request by ulfalizer**
See https://github.com/zephyrproject-rtos/zephyr/pull/13205.

Cleans up some binding (.yaml)-related stuff this time around.

### CODE REVIEW

File: scripts/dts/extract_dts_includes.py [hunk 1]
```diff
@@ -42,15 +43,19 @@ class Bindings(yaml.Loader):
...
+    def bindings(cls, dts_compats, yaml_dirs):
         # Find all .yaml files in yaml_dirs
         cls._files = []
         for yaml_dir in yaml_dirs:
             for root, dirnames, filenames in os.walk(yaml_dir):
                 for filename in fnmatch.filter(filenames, '*.yaml'):
                     cls._files.append(os.path.join(root, filename))
 
-        yaml_list = {'node': {}, 'bus': {}, 'compat': []}
+        compat_to_yaml = {}
```
Comment (b0661) [inline scripts/dts/extract_dts_includes.py]: Is the yaml format really of relevance here? Could be compat_to_binding instead.

File: scripts/dts/extract_dts_includes.py [hunk 2]
```diff
@@ -42,15 +43,19 @@ class Bindings(yaml.Loader):
...
+        # Maps buses to dictionaries that map compats to YAML nodes
+        bus_to_yaml = defaultdict(dict)
```
Comment (b0661) [inline scripts/dts/extract_dts_includes.py]: Same here - could be bus_to_binding.

File: scripts/dts/extract_dts_includes.py [hunk 3]
```diff
@@ -340,14 +340,9 @@ def dict_merge(parent, fname, dct, merge_dct):
...
-    """ Recursive overload procedure inside ``node``
-    ``inherits`` section is searched for and used as node base when found.
-    Base values are then overloaded by node values
-    and some consistency checks are done.
-    :param fname: initial yaml file being processed
-    :param node:
-    :return: node
-    """
+    # Recursively merges properties from files !include'd from the 'inherits'
+    # section of the binding. 'fname' is the path to the top-level YAML file,
```
Comment (b0661) [inline scripts/dts/extract_dts_includes.py]: YAML file -> bindings file

File: scripts/dts/extract_dts_includes.py [hunk 4]
```diff
@@ -103,15 +103,12 @@ def _include(self, node):
...
-                result.append(self._extract_file(filename))
-            return result
+        if isinstance(node, yaml.SequenceNode):
+            return [self._extract_file(fname)
+                    for fname in self.construct_sequence(node)]
 
-        else:
-            print("Error:: unrecognised node type in !include statement")
-            raise yaml.constructor.ConstructorError
+        print("Error:: unrecognised node type in !include statement")
```
Comment (b0661) [inline scripts/dts/extract_dts_includes.py]: '::' was a typo of me and still is a typo

**Review by b0661 — CHANGES_REQUESTED** [review]

**Comment by ulfalizer** [comment]
@b0661 
All fixed.

**Comment by ulfalizer** [comment]
@galak 
ping

**Review by nashif — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Pull request merged**